crates/stages/src/stages/execution.rs

```diff
@@ -263,6 +299,41 @@ impl<EF: ExecutorFactory> ExecutionStage<EF> {
     }
 }
 
+/// Temporary workaround trait for ExExManagerHandle not being importable in this crate due to a
+/// circular dependency.
+///
+/// The circular dependency is:
+///     reth-exex depends on reth-node-core for some config types which depends on reth-stages
+///     reth-stages depends on reth-exex for the handle
+///
+/// TODO(onbjerg): Remove this as soon as possible
+pub trait TempManagerHandle: Send + Sync + 'static {
+    /// Synchronously send a notification over the channel to all execution extensions.t.
+    fn send(
+        &self,
+        notification: CanonStateNotification,
+    ) -> Result<(), SendError<CanonStateNotification>>;
+
+    /// Get the current capacity of the ExEx manager's internal notification buffer.
+    fn capacity(&self) -> usize;
+
+    /// Returns `true` if there are ExEx's installed in the node.
+    fn has_exexs(&self) -> bool;
+
+    /// The finished height of all ExEx's.
+    ///
+    /// This is the lowest common denominator between all ExEx's. If an ExEx has not emitted a
+    /// `FinishedHeight` event, it will be `None`.
+    ///
+    /// This block is used to (amongst other things) determine what blocks are safe to prune.
+    ///
+    /// The number is inclusive, i.e. all blocks `<= finished_height` are safe to prune.
+    fn finished_height(&mut self) -> Option<BlockNumber>;
```

Comment:
do we need `capacity` and `finished_height` methods in the execution stage?

Reply:
no, these are leftovers - if we were going to send notifications 1 block at a time, we would use the capacity to inform us of when to stop a batch early

we don't need `finished_height` because the exex manager will simply discard notifications that no exex subscribes to

will rm